pennylane/_device.py

```diff
@@ -36,6 +36,48 @@
 from pennylane.wires import Wires, WireError
 
 
+def _process_shot_sequence(shot_list):
+    """Process the shot sequence, to determine the total
+    number of shots and the shot vector.
+
+    Args:
+        shot_list (Sequence[int, tuple[int]]): sequence of non-negative shot integers
```

Comment:
At the back of my mind, I wanted to allow (advanced) users and other parts of the codebase to pass `shots=[(1, 1e9)]`, rather than having to do `shots=[1] * 1000000000`, which will create a large list in memory for absolutely no reason